### PR TITLE
gh-114275: Skip doctests that use `asyncio` in `test_pdb` for WASI builds

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-114275

### CODE REVIEW

**Comment by sobolevn** [comment]
!buildbot wasi*

File: Lib/test/test_pdb.py [hunk 1]
```diff
@@ -1693,122 +1693,123 @@ def test_pdb_next_command_for_generator():
...
-    ...     import pdb; pdb.Pdb(nosigint=True, readrc=False).set_trace()
-    ...     await test_coro()
-
-    >>> def test_function():
-    ...     loop = asyncio.new_event_loop()
-    ...     loop.run_until_complete(test_main())
-    ...     loop.close()
-    ...     asyncio.set_event_loop_policy(None)
-    ...     print("finished")
+if support.has_socket_support:
```
Comment (vstinner) [inline Lib/test/test_pdb.py]: I suggest to add a constant with a comment to explain the skip in a comment. Example:

```py
# gh-114275: WASI fails to run asyncio tests, similar skip than test_asyncio.
SKIP_ASYNCIO_TESTS = (not support.has_socket_support)

(...)

if not SKIP_ASYNCIO_TESTS: (...)
```

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by vstinner** [comment]
!buildbot WASI

**Comment by vstinner** [comment]
I prefer to be extra safe and check again the updated PR on WASI buildbots.

**Comment by vstinner** [comment]
Tests passed on the two WASI buildbot workers. I merged your fix, thanks.

**Pull request merged**